src/datastore/pg-store-v2.ts

```diff
@@ -269,4 +294,152 @@ export class PgStoreV2 extends BasePgStoreModule {
+          LIMIT ${limit}
+          OFFSET ${offset}
+        )
+        SELECT
+          ${sql(TX_COLUMNS)},
+          (
+            SELECT COALESCE(SUM(amount), 0)
+            FROM stx_events
+            WHERE ${eventCond} AND sender = ${args.address}
+          ) + txs.fee_rate AS stx_sent,
+          (
+            SELECT COALESCE(SUM(amount), 0)
+            FROM stx_events
+            WHERE ${eventCond} AND recipient = ${args.address}
+          ) AS stx_received,
+          (
+            SELECT COUNT(*)::int FROM stx_events
+            WHERE ${eventAcctCond} AND asset_event_type_id = ${DbAssetEventTypeId.Transfer}
+          ) AS stx_transfer,
+          (
+            SELECT COUNT(*)::int FROM stx_events
+            WHERE ${eventAcctCond} AND asset_event_type_id = ${DbAssetEventTypeId.Mint}
+          ) AS stx_mint,
+          (
+            SELECT COUNT(*)::int FROM stx_events
+            WHERE ${eventAcctCond} AND asset_event_type_id = ${DbAssetEventTypeId.Burn}
+          ) AS stx_burn,
+          (
+            SELECT COUNT(*)::int FROM ft_events
+            WHERE ${eventAcctCond} AND asset_event_type_id = ${DbAssetEventTypeId.Transfer}
+          ) AS ft_transfer,
+          (
+            SELECT COUNT(*)::int FROM ft_events
+            WHERE ${eventAcctCond} AND asset_event_type_id = ${DbAssetEventTypeId.Mint}
+          ) AS ft_mint,
+          (
+            SELECT COUNT(*)::int FROM ft_events
+            WHERE ${eventAcctCond} AND asset_event_type_id = ${DbAssetEventTypeId.Burn}
+          ) AS ft_burn,
+          (
+            SELECT COUNT(*)::int FROM nft_events
+            WHERE ${eventAcctCond} AND asset_event_type_id = ${DbAssetEventTypeId.Transfer}
+          ) AS nft_transfer,
+          (
+            SELECT COUNT(*)::int FROM nft_events
+            WHERE ${eventAcctCond} AND asset_event_type_id = ${DbAssetEventTypeId.Mint}
+          ) AS nft_mint,
+          (
+            SELECT COUNT(*)::int FROM nft_events
+            WHERE ${eventAcctCond} AND asset_event_type_id = ${DbAssetEventTypeId.Burn}
+          ) AS nft_burn,
+          count
+        FROM stx_txs
+        INNER JOIN txs USING (tx_id, index_block_hash, microblock_hash)
+      `;
+      const total = resultQuery.length > 0 ? resultQuery[0].count : 0;
+      const parsed = resultQuery.map(r => parseAccountTransferSummaryTxQueryResult(r));
+      return {
+        total,
+        limit,
+        offset,
+        results: parsed,
+      };
+    });
+  }
+
+  async getAddressTransactionEvents(
+    args: AddressTransactionParams & TransactionPaginationQueryParams
+  ): Promise<DbPaginatedResult<DbAddressTransactionEvent>> {
+    return await this.sqlTransaction(async sql => {
+      await assertAddressExists(sql, args.address);
+      await assertTxIdExists(sql, args.tx_id);
+      const limit = args.limit ?? TransactionLimitParamSchema.default;
+      const offset = args.offset ?? 0;
+
+      const eventCond = sql`
```

Comment:
are there indexes for sender and recipient?

Reply:
Yep, we're good 👍